checkov/secrets/plugins/detector_utils.py

```diff
@@ -0,0 +1,252 @@
+from __future__ import annotations
+
+import json
+import re
+from re import Pattern
+from typing import Any, TYPE_CHECKING
+
+from detect_secrets.util.filetype import FileType
+from detect_secrets.plugins.keyword import DENYLIST
+from detect_secrets.plugins.keyword import AFFIX_REGEX
+from detect_secrets.plugins.keyword import CLOSING
+from detect_secrets.plugins.keyword import OPTIONAL_WHITESPACE
+from detect_secrets.plugins.keyword import QUOTE
+from detect_secrets.plugins.keyword import SECRET
+
+from checkov.secrets.parsers.terraform.multiline_parser import terraform_multiline_parser
+from checkov.secrets.parsers.terraform.single_line_parser import terraform_single_line_parser
+from checkov.secrets.parsers.yaml.multiline_parser import yml_multiline_parser
+from checkov.secrets.parsers.json.multiline_parser import json_multiline_parser
+
+if TYPE_CHECKING:
+    from checkov.secrets.parsers.multiline_parser import BaseMultiLineParser
+    from detect_secrets.core.potential_secret import PotentialSecret
+    from detect_secrets.util.code_snippet import CodeSnippet
+    from detect_secrets.plugins.high_entropy_strings import HighEntropyStringsPlugin
+
+MAX_KEYWORD_LIMIT = 500
+
+DENY_LIST_REGEX = r'|'.join(DENYLIST)
```

Comment:
maybe all the constants here should be in a separate class for keyword_combinator? (maybe also the `analyze_multiline` method?) as they are not general detector utils for secrets as general?

Reply:
The issue is that the extraction is happening to open these directives for usage in other places.
In general, any non-regex detector should be able to use them as they are generic and aren't targeted at keyword combinator.